channels/layers.py

```diff
@@ -363,7 +372,9 @@ async def group_discard(self, group, channel):
     async def group_send(self, group, message):
         # Check types
         assert isinstance(message, dict), "Message is not a dict"
-        assert self.valid_group_name(group), "Invalid group name"
+        assert self.valid_group_name(group), (
+            f"Group name must be" f"less than {self.MAX_NAME_LENGTH} characters."
+        )
```

Comment:
As per the point about the `TypeError` being raised before the `AssertionError`, it's not clear we need to make any change here, or on line 353/360 above.